lib/msf/core/exploit/remote/kerberos/auth_brute.rb

```diff
@@ -61,17 +61,29 @@ def attempt_kerberos_logins
 
       case result.status
       when Metasploit::Model::Login::Status::SUCCESSFUL
-        hash = format_as_rep_to_john_hash(proof.as_rep)
-
-        # Accounts that have 'Do not require Kerberos preauthentication' enabled, will receive an ASREP response with a
-        # ticket present without requiring a password
-        if !proof.preauth_required
-          print_good("#{peer} - User: #{user.inspect} does not require preauthentication. Hash: #{hash}")
+        case proof
+        when Rex::Proto::Kerberos::Model::Error::KerberosError
+        if proof.error_code == Rex::Proto::Kerberos::Model::Error::ErrorCodes::KDC_ERR_KEY_EXPIRED || proof.error_code == Rex::Proto::Kerberos::Model::Error::ErrorCodes::KRB_AP_ERR_SKEW
+          print_good("#{peer} - User found: #{user.inspect} with password #{password}, but no ticket received (#{proof})")
+          report_cred(user: user, password: password)
         else
-          print_good("#{peer} - User found: #{user.inspect} with password #{password}. Hash: #{hash}")
+          raise ::RuntimeError, 'Successful login reported, but unknown error'
         end
+        when Msf::Exploit::Remote::Kerberos::Model::TgtResponse
+
+          hash = format_as_rep_to_john_hash(proof.as_rep)
 
-        report_cred(user: user, password: password, asrep: hash)
+          # Accounts that have 'Do not require Kerberos preauthentication' enabled, will receive an ASREP response with a
+          # ticket present without requiring a password
+          if !proof.preauth_required
+            print_good("#{peer} - User: #{user.inspect} does not require preauthentication. Hash: #{hash}")
+          else
+            print_good("#{peer} - User found: #{user.inspect} with password #{password}. Hash: #{hash}")
+          end
+          report_cred(user: user, password: password, asrep: hash)
+        else
+          raise ::RuntimeError, 'Successful login reported, but unknown proof type'
```

Comment:
Same comment regarding logging the credential data and not raising an exception.